src/utils/flags.js

```diff
@@ -0,0 +1,75 @@
+import { flags as flagsConfig } from '../config';
+
+/**
+ * Flag definitions
+ */
+export const Flags = {
+  /**
+   * Returns `true` if the flag is defined otherwise `false`
+   * @param {string} name The flag name to test
+   * @returns {boolean} The result
+   */
+  isDefined: name => Flags.names().includes(name),
+
+  /**
+   * Returns an array of defined flag names
+   * @returns {array} The defined flag names
+   */
+  names: () => Object.keys(flagsConfig),
+
+  /**
+   * Returns an object mapping flag names to their default values
+   * @returns {object} The defined flag defaults
+   */
+  defaults: () =>
+    Flags.names().reduce(
+      (result, flag) =>
+        Object.assign(result, { [flag]: flagsConfig[flag].default }),
+      {}
+    )
+};
+
+/**
+ * Returns an object with flags as set in given or current URL
+ * @param {string=} url The URL (optional, default current location)
+ * @returns {object} An object with flags and their values
+ */
+export const getFlagsFromUrl = url => {
+  const urlParams = new URL(url || document.location).searchParams;
+  const flags = {};
+
+  [...urlParams].forEach(([name, value]) =>
+    Flags.isDefined(name)
+      ? (flags[name] = value === 'true' || value === '1')
```

Comment:
I feel like it should probably check for `value === ''` and evaluate that to `true` too, no? Because I tend to regard valueless query string keys as being truthy even though technically their value is `''`, and I feel like it should be okay to just add `?newgraph` to turn the feature on. What do you think?

Reply:
Good catch, I've added this one too, thanks.